src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,33 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    public static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+    public String getSalaryInfo(String[] names, String[] data,
+                                String dateFrom, String dateTo) {
+        StringBuilder report = new StringBuilder().append("Report for period ")
+                .append(dateFrom).append(" - ").append(dateTo);
+        LocalDate localDateFrom = LocalDate.parse(dateFrom, formatter);
+        LocalDate localDateTo = LocalDate.parse(dateTo, formatter);
+        for (String name : names) {
+            int salary = 0;
+            for (String line: data) {
+                String[] dataParts = line.split(" ");
+                LocalDate localDate = LocalDate.parse(dataParts[0], formatter);
+                if ((localDate.isAfter(localDateFrom) || localDate.equals(localDateFrom))
+                        && (localDate.isBefore(localDateTo)
+                        || localDate.equals(localDateTo)) && dataParts[1].equals(name)) {
```

Comment:
This condition is a bit complex and hard to read. Consider breaking it down into smaller conditions and using descriptive variable names for them. This will make your code easier to understand.